src/server/standalone.js

```diff
@@ -584,9 +585,33 @@ function StandAloneServer(gmeConfig) {
 
     setupExternalRestModules();
 
-    // Basic authentication
+    __app.get(['', '/', '/index.html'], ensureAuthenticated, function (req, res) {
+        var indexHtmlPath = Path.join(__clientBaseDir, 'index.html'),
+            protocol = gmeConfig.server.behindSecureProxy ? 'https' : 'http',
+            host = protocol + '://' + req.get('host'),
+            url = host + req.originalUrl,
+            imageUrl = host + '/img/gme-logo.png',
+            projectId = req.query.project;
+
+        logger.debug('resolved url', url);
+
+        fs.readFile(indexHtmlPath, 'utf8', function (err, data) {
+            if (err) {
+                logger.error(err);
+                res.send(404);
+            } else {
+                res.contentType('text/html');
+                res.send(
+                    data.replace(/__URL__/g, url)
+                        .replace(/__IMAGE_URL__/g, imageUrl)
+                        .replace(/__TITLE__/g, projectId ? projectId.replace('+', '/') : 'WebGME')
```

Comment:
Need to escape HTML entities here. Current code has a XSS vulnerability (e.g. set projectId to `"><script>...evil stuff...</script><a href="`)